src/controllers/SelectPenaltyController.ts

```diff
@@ -55,7 +56,8 @@ class Processor implements FormActionProcessor {
 }
 
 // tslint:disable-next-line: max-classes-per-file
-@controller(SELECT_THE_PENALTY_PAGE_URI, SessionMiddleware, AuthMiddleware, PenaltyReferenceRouter)
+@controller(SELECT_THE_PENALTY_PAGE_URI, SessionMiddleware, AuthMiddleware, PenaltyReferenceRouter,
```

Comment:
I do not think `CompanyAuthMiddleware` will be invoked when `PenaltyReferenceRouter` did redirect to `REVIEW_PENALTY_PAGE_URI`, although I might be always wrong.

Despite above we need at least few tests that make sure that:
- auth code is enforced when legacy penalty ref is used and only one penalty was found
- auth code is enforced when legacy penalty ref is used and more than one penalty was found
- no auth code is enforced for non legacy penalty ref